### ISSUE TITLE
LocalCopyPropagation loses source position information

### ISSUE THREAD

**Issue by mbudiu-vmw** [p4lang/p4c#2678]
When copy propagating an expression we should use the source position information of the destination expression. Otherwise errors are hard to read.

**Comment by jafingerhut** [comment]
I see this issue with the 'fixed' label, and I see a link to a branch on Mihai's forked repo, but I did not find a PR, or evidence that it has been merged in.  Was that PR ready?

**Comment by mbudiu-vmw** [comment]
I don't remember which PR fixed this issue, it was bundled into a PR for other problems.

**Comment by mbudiu-vmw** [comment]
Yes, that is the fix: https://github.com/mbudiu-vmw/p4c-clone/commit/d22c2a2ec565c8aaa0b400e4705c064c72ccede8

**Comment by jafingerhut** [comment]
That commit mbudiu-vmw@d22c2a2  adds a file copySrcInfo.h which does not exist anywhere in the latest p4c.

**Comment by mbudiu-vmw** [comment]
I guess it is part of the experimental support for general-purpose safe unions which hasn't been merged.
I should pull it out as a separate PR and submit it.
